### PR TITLE
[AC-7805] Cancellation workflow for office hours patch

### PR DESCRIPTION
**Pull request by iankigen**
## [AC-7805](https://masschallenge.atlassian.net/browse/AC-7805)

**Changes introduced**
- allow mentors to cancel reserved office hour sessions

**Testing**
- checkout to this branch and `make run-server-1`
- As a mentor with a reserved office hour session visit [cancel office hour endpoint](http://localhost:8000/api/v1/cancel_office_hour_session/). 
Allowed methods = [POST]
- POST data format:
```
 {
    'id': <office hour session id>,
    'message': <optional message to attendees>
}
```
Confirm that:

- A mentor can cancel reserved office hour session:
- Email notification is sent to the **office hours holder’s** and **finalist's** email address when a reserved session is canceled by the office hour holder
  - including the message to attendees if provided

[Accelerate sibling PR](https://github.com/masschallenge/accelerate/pull/2629)

### CODE REVIEW

**Review by geofrocker — APPROVED** [review]
QA looks good ✅
- Tested cancellation of a reserved hour by the office hour holder ie user_id=>1032 ✅
- Tested cancellation of an open hour by the office hour holder ie user_id=>1032 ✅
- Tested that email respective email are sent on successful cancellation ✅
- Tested cancellation of a non-existent office hour ✅
- Email is sent to a finalist on successful cancellation ✅
I am not sure about this wording in the email sent to the finalist ✔️ `Kevin Winter cancelled your office hours session with Kevin winter.` cc @hfreund

**Comment by hfreund** [comment]
>       I am not sure about this wording in the email sent to the finalist ✔️ `Kevin Winter cancelled your office hours session with Kevin winter.` cc @hfreund

@geofrocker I don't see any reference to Kevin Winter in the code, nor should there be. The email should be specific to the mentor who canceled / owns the office hour.

**Comment by geofrocker** [comment]
> > ```
> >   I am not sure about this wording in the email sent to the finalist ✔️ `Kevin Winter cancelled your office hours session with Kevin winter.` cc @hfreund
> > ```
> 
> @geofrocker I don't see any reference to Kevin Winter in the code, nor should there be. The email should be specific to the mentor who canceled / owns the office hour.

This is just an example of how the email will look like if `Kevin Winter` cancelled a finalist's office hour session.

**Comment by hfreund** [comment]
> > > ```
> > >   I am not sure about this wording in the email sent to the finalist ✔️ `Kevin Winter cancelled your office hours session with Kevin winter.` cc @hfreund
> > > ```
> > 
> > 
> > @geofrocker I don't see any reference to Kevin Winter in the code, nor should there be. The email should be specific to the mentor who canceled / owns the office hour.
> 
> This is just an example of how the email will look like if `Kevin Winter` cancelled a finalist's office hour session.

@geofrocker @iankigen  I believe the text of this email is defined on a separate ticket: https://masschallenge.atlassian.net/browse/AC-7562?focusedCommentId=43035
Can you confirm that is true?

**Comment by iankigen** [comment]
@hfreund [AC-7562](https://masschallenge.atlassian.net/browse/AC-7562?focusedCommentId=43035) email templates are used when a **finalist** cancels a reservation. 
The email template in question is used when **mentor** cancels a session(reserved/unreserved). It was not defined in the [ticket](https://masschallenge.atlassian.net/browse/AC-7561) and therefore we used the existing email template:
```
{{ cancelled_by }} canceled your office hours session{{ hours_with }}.
When: {{ date }}, {{ start_time }} (CANCELED)
Where: {{ location }}
{% if message %}
    Message: {{ message }}
{% endif %}
You may check the status of your schedule at any time by visiting your Dashboard:
{{ dashboard_url }} (log in using: {{ dashboard_username }}).
If you have questions, don't hesitate to contact us at any time via
<a href="https://masschallenge.org/contact">
    https://masschallenge.org/contact</a>
- The MassChallenge Team 
{{ phone }}
www.masschallenge.org

File: web/impact/impact/v1/views/cancel_office_hour_session_view.py [hunk 1]
```diff
@@ -64,22 +63,31 @@ def get_response(success, detail):
     })
 
 
+def get_cancelled_by(mentor_name=None, staff=False):
+    to_finalist, to_mentor = mentor_name, 'You have'
```
Comment (jonkiparsky) [inline web/impact/impact/v1/views/cancel_office_hour_session_view.py:67]: I don't love this shorthand style of writing assignments, but won't block on this.

**Review by jonkiparsky — APPROVED** [review]
I feel like this is crying out for a refactor, but will approve it in the interests of getting on with the work.

**Pull request merged**